src/main/java/com/marklogic/rest/util/RestConfig.java

```diff
@@ -37,12 +44,50 @@ public RestConfig(RestConfig other) {
 		if (other.scheme != null) {
 			this.scheme = other.scheme;
 		}
+
 		this.configureSimpleSsl = other.configureSimpleSsl;
 		this.useDefaultKeystore = other.useDefaultKeystore;
 		this.sslProtocol = other.sslProtocol;
 		this.trustManagementAlgorithm = other.trustManagementAlgorithm;
 		this.sslContext = other.sslContext;
 		this.hostnameVerifier = other.hostnameVerifier;
+		this.sslHostnameVerifier = other.sslHostnameVerifier;
+	}
+
+	public DatabaseClientBuilder newDatabaseClientBuilder() {
+		DatabaseClientBuilder builder = new DatabaseClientBuilder()
```

Comment:
@BillFarber This is the main addition - `RestConfig` is used by ml-app-deployer to carry around connection and auth info. This method knows how to apply the inputs in a `RestConfig` instance in order to construct a `DatabaseClientBuilder`, which `RestTemplateUtil` can then use to construct an OkHttp Client.